core/sr-primitives/src/generic/checked_extrinsic.rs

```diff
@@ -75,25 +75,25 @@ where
 	}
 
 	fn dispatch(self,
-		weight: crate::weights::Weight,
+		info: TransactionInfo,
 		len: usize,
 	) -> Result<DispatchResult, DispatchError> {
 		let maybe_who = if let Some((id, extra)) = self.signed {
-			Extra::pre_dispatch(extra, &id, weight, len)?;
+			Extra::pre_dispatch(extra, &id, info, len)?;
 			Some(id)
 		} else {
-			Extra::pre_dispatch_unsigned(weight, len)?;
+			Extra::pre_dispatch_unsigned(info, len)?;
 			None
 		};
 		Ok(self.function.dispatch(Origin::from(maybe_who)))
 	}
 }
 
-impl<AccountId, Call, Extra> Weigh for CheckedExtrinsic<AccountId, Call, Extra>
+impl<AccountId, Call, Extra> DispatchInfo for CheckedExtrinsic<AccountId, Call, Extra>
```

Comment:
probably need to rename `DispatchInfo` to `GetDispatchInfo` and `dispatch_info` to `get_dispatch_info` so that `TransactionInfo` (which is misnamed, since it's not necessarily about a transaction) can be `DispatchInfo`.